src/main/java/io/zipcoder/Classroom.java

```diff
@@ -1,4 +1,143 @@
 package io.zipcoder;
 
+import java.util.Arrays;
+import java.util.HashMap;
+
 public class Classroom {
+
+    protected Student[] students;
+    HashMap<Student, String> gradeBook= new HashMap<>();
+
+    public Classroom() {
+
+        this.students = new Student[30];
+    }
+
+    public Classroom(int maxNumberOfStudents) {
+        this.students = new Student[maxNumberOfStudents];
+    }
+
+    public Classroom(Student[] aStudentArray) {
+        this.students = aStudentArray;
+    }
+
```

Comment:
you could nest these constructors a little better.

Reply:
I'm not sure exactly how.  I may try to grab you for a quick second if I see you on the floor.